cluster/juju/layers/kubernetes-worker/templates/ingress-daemon-set.yaml

```diff
@@ -94,61 +144,72 @@ rules:
       - endpoints
     verbs:
       - get
-      - create
-      - update
 ---
 apiVersion: rbac.authorization.k8s.io/v1beta1
 kind: RoleBinding
 metadata:
   name: nginx-ingress-role-nisa-{{ juju_application }}-binding
+  labels:
+    app.kubernetes.io/name: ingress-nginx
+    app.kubernetes.io/part-of: ingress-nginx
+    cdk-{{ juju_application }}-ingress: "true"
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: Role
   name: nginx-ingress-{{ juju_application }}-role
 subjects:
   - kind: ServiceAccount
     name: nginx-ingress-{{ juju_application }}-serviceaccount
+    namespace: ingress-{{ juju_application }}-nginx
 ---
 apiVersion: rbac.authorization.k8s.io/v1beta1
 kind: ClusterRoleBinding
 metadata:
   name: nginx-ingress-clusterrole-nisa-{{ juju_application }}-binding
+  labels:
+    app.kubernetes.io/name: ingress-nginx
+    app.kubernetes.io/part-of: ingress-nginx
+    cdk-{{ juju_application }}-ingress: "true"
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
   name: nginx-ingress-{{ juju_application }}-clusterrole
 subjects:
   - kind: ServiceAccount
     name: nginx-ingress-{{ juju_application }}-serviceaccount
-    namespace: default
----
-apiVersion: v1
-kind: ConfigMap
-metadata:
-  name: nginx-load-balancer-{{ juju_application }}-conf
+    namespace: ingress-{{ juju_application }}-nginx
 ---
 apiVersion: {{ daemonset_api_version }}
 kind: DaemonSet
 metadata:
   name: nginx-ingress-{{ juju_application }}-controller
+  namespace: ingress-{{ juju_application }}-nginx
   labels:
+    app.kubernetes.io/name: ingress-nginx
+    app.kubernetes.io/part-of: ingress-nginx
     juju-application: nginx-ingress-{{ juju_application }}
+    cdk-{{ juju_application }}-ingress: "true"
 spec:
   selector:
     matchLabels:
-      name: nginx-ingress-{{ juju_application }}
+      app.kubernetes.io/name: ingress-nginx
+      app.kubernetes.io/part-of: ingress-nginx
```

Comment:
I think we need to keep `{{ juju_application }}` somewhere in these selector labels and the corresponding pod template labels. Otherwise, I would expect problems when you have 2 kubernetes-worker applications and 2 DaemonSets, as the 2 DaemonSets try to select and manage the same set of pods.